runner.go

```diff
@@ -152,6 +152,16 @@ type passThrough struct {
 	ReturnTypes []Type
 }
 
+func (r *passThrough) Scopes() StringsSet {
+	scopes := make(StringsSet)
+	for _, t := range r.ReturnTypes {
+		if s := GetScope(t); s != "" {
+			scopes[s] = struct{}{}
```

Comment:
what do you think about extending StringSet api?
```suggestion
			scopes.Add(s)
```